src/js_realm_object.hpp

```diff
@@ -140,6 +142,50 @@ bool RealmObjectClass<T>::set_property(ContextType ctx, ObjectType object, const
     return true;
 }
 
+template<typename T>
+void RealmObjectClass<T>::set_link(ContextType ctx, ObjectType object, Arguments args, ReturnValue& return_value) {
+    args.validate_count(2);
+
+    auto realm_object = get_internal<T, RealmObjectClass<T>>(object);
+    realm_object->realm()->verify_in_write();
+
+    NativeAccessor<T> accessor(ctx, realm_object->realm(), realm_object->get_object_schema());
+    std::string property_name = Value::validated_to_string(ctx, args[0], "propertyName");
+    const Property* prop = realm_object->get_object_schema().property_for_name(property_name);
+    if (!prop) {
+        throw std::invalid_argument(util::format("No such property: %1", property_name));
+    }
+    if (prop->type != realm::PropertyType::Object) {
+        throw TypeErrorException(accessor, realm_object->get_object_schema().name, *prop, args[1]);
+    }
+    auto& linked_schema = *realm_object->realm()->schema().find(prop->object_type);
+    auto linked_pk = linked_schema.primary_key_property();
+    if (!linked_pk) {
+        throw std::invalid_argument("Linked object type must have a primary key.");
+    }
+
+    auto table = realm_object->row().get_table();
+    auto linked_table = table->get_link_target(prop->table_column);
+
+    size_t row_ndx = realm::not_found;
+    if (linked_pk->type == realm::PropertyType::String) {
+        row_ndx = linked_table->find_first(linked_pk->table_column,
+                                           accessor.template unbox<StringData>(args[1]));
+    }
+    else if (is_nullable(linked_pk->type)) {
+        row_ndx = linked_table->find_first(linked_pk->table_column,
+                                           accessor.template unbox<util::Optional<int64_t>>(args[1]));
+    }
+    else {
+        row_ndx = linked_table->find_first(linked_pk->table_column,
+                                           accessor.template unbox<int64_t>(args[1]));
+    }
+
+    if (row_ndx != realm::not_found) {
+        realm_object->row().set_link(prop->table_column, row_ndx);
+    }
```

Comment:
Should we throw an exception here?

Reply:
The object not existing is valid (it may have been excluded by a filter), but it probably should set_null() in that case.